src/EventStore.Core/ClusterVNode.cs

```diff
@@ -1390,7 +1390,7 @@ public class ClusterVNode<TStreamId> :
 			}
 
 			var scavengerLogManager = new TFChunkScavengerLogManager(
-				nodeEndpoint: NodeInfo.HttpEndPoint.ToString(),
+				nodeEndpoint: _options.Interface.ReplicationHostAdvertiseAs == null ? NodeInfo.HttpEndPoint.ToString() : (_options.Interface.ReplicationHostAdvertiseAs + ":" + NodeInfo.HttpEndPoint.GetPort()),
```

Comment:
You should use the `GossipAdvertiseInfo` here rather than the `NodeInfo`, since that will already have the best host and port according to configuration:

```suggestion
				nodeEndpoint: $"{GossipAdvertiseInfo.HttpEndPoint.Host}:{GossipAdvertiseInfo.HttpEndPoint.Port}",
```
Don't use `GossipAdvertiseInfo.HttpEndPoint.ToString()` directly though, because that will try and add the address family and format it as `Unspecified/{hostname}:{port}`

Reply:
Thanks for the review, changes done.